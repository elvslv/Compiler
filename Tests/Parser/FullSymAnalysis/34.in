var i, j: integer;
r: real;
function func(i: integer): integer;
begin
	func := i;
end;
begin
	while func(i) do
		i := i + 1;
end.