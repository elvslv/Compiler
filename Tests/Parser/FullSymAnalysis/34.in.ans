var I : INTEGER;
var J : INTEGER;
var R : REAL;

function FUNC(var I : INTEGER;): INTEGER;

begin       
|           
|___:=      
|   |       
|   |___FUNC
|   |       
|   |___I   
|           
end         


begin                
|                    
|___while            
|   |                
|   |___()           
|   |   |            
|   |   |___FUNC     
|   |   |            
|   |   |___I        
|   |                
|   do               
|   |                
|   |___begin        
|       |            
|       |___:=       
|       |   |        
|       |   |___I    
|       |   |        
|       |   |___+    
|       |       |    
|       |       |___I
|       |       |    
|       |       |___1
|       |            
|       end          
|                    
end                  
