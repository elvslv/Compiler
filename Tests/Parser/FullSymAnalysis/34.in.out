function FUNC(var I : INTEGER;): INTEGER;
begin         
|             
|___:=        
|   |         
|   |___RESULT
|   |         
|   |___I     
|             
end           

var I : INTEGER;
var J : INTEGER;
var R : REAL;

begin                
|                    
|___while            
|   |                
|   |___()           
|   |   |            
|   |   |___FUNC     
|   |   |            
|   |   |___I        
|   |                
|   do               
|   |                
|   |___begin        
|       |            
|       |___:=       
|       |   |        
|       |   |___I    
|       |   |        
|       |   |___+    
|       |       |    
|       |       |___I
|       |       |    
|       |       |___1
|       |            
|       end          
|                    
end                  
